crates/spirv-std/src/storage_class.rs

```diff
@@ -0,0 +1,193 @@
+//! # Storage Classes
+//!
+//! Class of storage for declared variables. Intermediate values do not form a
+//! storage class, and unless stated otherwise, storage class-based restrictions
+//! are not restrictions on intermediate objects and their types.
```

Comment:
Pointing out that they're aliases for pointers is important I think. I've seen some confusion over this in the rust-gpu discord, I don't think it's very clear right now that "`Input<T>`" means "a *pointer* to T, where that pointer is of style "input"" - and considering storage classes are literally "pointer flavors", seems important to document.